src/pretix/control/views/event.py

```diff
@@ -399,6 +403,56 @@ def post(self, request, *args, **kwargs):
             return self.get(request)
 
 
+class MailSettingsPreview(EventPermissionRequiredMixin, View):
+    permission = 'can_change_settings'
+
+    # return the origin text if key is missing in dict
+    class SafeDict(dict):
+        def __missing__(self, key):
+            return '{' + key + '}'
+
+    @cached_property
+    def items(self):
+        return ['mail_text_order_placed', 'mail_text_order_paid', 'mail_text_order_free', 'mail_text_resend_link',
+                'mail_text_resend_all_links', 'mail_text_order_changed', 'mail_days_order_expire_warning',
+                'mail_text_order_expire_warning', 'mail_text_waiting_list']
+
+    @cached_property
+    def dummy_data(self):
+        return {
+            'mail_text_order_placed': {
+                'event': str(self.request.event.name),
+                'total': 100,
+                'currency': self.request.event.currency,
+                'date': date_format(now() + timedelta(days=7), 'SHORT_DATE_FORMAT'),
+                'paymentinfo': _('Please transfer money to this bank account: AXXXX1033'),
+                'url': build_absolute_uri('presale:event.index', kwargs={
+                    'event': self.request.event,
+                    'organizer': self.request.event.organizer
+                }),
+                'invoice_name': _('Invoice for %s.') % {str(self.request.event.name)},
+                'invoice_company': str(self.request.event.organizer.name)
+            }
+        }
+
+    def post(self, request, *args, **kwargs):
+        preview_item = request.POST.get('item', '')
+        if preview_item not in self.items:
+            raise HttpResponseBadRequest(_('invalid item'))
+
+        regex = r"^" + re.escape(preview_item) + r"_\d{1}$"
```

Comment:
I don't think it's a good idea to hardcode that we'll never have more than 10 languages in an event (although it is unlikely, I agree).